numpy/lib/nanfunctions.py

```diff
@@ -606,6 +609,145 @@ def nanprod(a, axis=None, dtype=None, out=None, keepdims=np._NoValue):
     return np.prod(a, axis=axis, dtype=dtype, out=out, keepdims=keepdims)
 
 
+def nancumsum(a, axis=None, dtype=None, out=None):
+    """
+    Return the cumulative sum of array elements over a given axis treating Not a
+    Numbers (NaNs) as zero.
+
+    One is returned for slices that are all-NaN or empty.
+
+    .. versionadded:: 1.11.0
```

Comment:
We're already at the release candidate stage for 1.11, so this will make it into 1.12